### PR TITLE
refactor: API refactor for package re-use

### LINKED ISSUE #289: feat: refactor ls command
**Describe the solution you'd like:**
Change ls command flags
Change current flags:
--alias 
--cluster-provider
--id string
--identity-provider
--provider-id

To:
--filter 

Example
--filter alias=\*dev*,cluster-provider=eks,identity=saml

**Why do you want this feature:**
This will make the command easier to use and more extensible. It also allows for filtering on flags in history entries
e.g. --filter namespace=dev1

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:
This is work in progress for a refcator to make kconnect useable as a package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #289

### CODE REVIEW

File: pkg/app/to.go [hunk 1]
```diff
@@ -56,16 +57,31 @@ func (a *App) ConnectTo(params *ConnectToParams) error {
...
-	if err != nil {
-		return fmt.Errorf("getting identity provider %s: %w", entry.Spec.Identity, err)
-	}
-	clusterProvider, err := provider.GetClusterProvider(entry.Spec.Provider)
-	if err != nil {
-		return fmt.Errorf("getting cluster provider %s: %w", entry.Spec.Provider, err)
-	}
-
-	cs, err := a.buildConnectToConfig(idProvider, clusterProvider, entry)
+	// idProviderInput := &provider.PluginCreationInput{
```
Comment (seankelly001) [inline pkg/app/to.go:60]: Lots of commented out code here

File: pkg/app/use.go [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+	return output.Cluster, nil
+}
+
+func (a *App) filterConfig(params *UseInput) map[string]string {
+	filteredConfig := make(map[string]string)
+
+	for _, configItem := range params.ConfigSet.GetAll() {
+
+		if configItem.Sensitive {
```
Comment (seankelly001) [inline pkg/app/use.go:202]: Could combine these 2 if statements

File: pkg/app/to.go (hunk 1 above)
Comment (richardcase) [reply pkg/app/to.go]: Thanks for finding this. Now removed

File: pkg/app/use.go (hunk 2 above)
Comment (richardcase) [reply pkg/app/use.go]: Combined the 2

**Review by seankelly001 — APPROVED** [review]
LGTM

**Pull request merged**